salt/modules/virt.py

```diff
@@ -673,6 +634,117 @@ def init(name,
                     )
                     define_vol_xml_str(xml)
 
+            elif hypervisor in ['qemu', 'kvm']:
+
```

Comment:
I'd say stick the ` for disk in diskp:` somewhere here; only loop when using the `libvirt` hypervisor.
Sticking lines 639-740 in a new function might make this a lot cleaner as well.